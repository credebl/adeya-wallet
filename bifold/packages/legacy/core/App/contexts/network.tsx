import { NetInfoStateType, useNetInfo } from '@react-native-community/netinfo'
import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import NetInfoModal from '../components/modals/NetInfoModal'
import { fetchLedgerNodes, canConnectToLedgerNode } from '../utils/ledger'

export interface NetworkContext {
  silentAssertConnectedNetwork: () => boolean
  assertConnectedNetwork: () => boolean
  displayNetInfoModal: () => void
  hideNetInfoModal: () => void
  assertLedgerConnectivity: () => Promise<boolean>
}

export const NetworkContext = createContext<NetworkContext>(null as unknown as NetworkContext)

export const NetworkProvider: React.FC = ({ children }) => {
  const netInfo = useNetInfo()
  const [isNetInfoModalDisplayed, setIsNetInfoModalDisplayed] = useState<boolean>(false)

  const displayNetInfoModal = () => {
    setIsNetInfoModalDisplayed(true)
  }

  const hideNetInfoModal = () => {
    setIsNetInfoModalDisplayed(false)
  }

  const silentAssertConnectedNetwork = () => {
    return netInfo.isConnected || netInfo.type !== NetInfoStateType.none
  }

  const assertConnectedNetwork = () => {
    const isConnected = silentAssertConnectedNetwork()
    if (!isConnected) {
      displayNetInfoModal()
    }

    return isConnected
  }

  const assertLedgerConnectivity = async (): Promise<boolean> => {
    const nodes = fetchLedgerNodes()

    if (typeof nodes === 'undefined' || nodes.length === 0) {
      return false
    }

    const connections = await Promise.all(nodes.map((n: { host: string; port: number }) => canConnectToLedgerNode(n)))
    return connections.includes(true)
  }

  return (
    <NetworkContext.Provider
      value={{
        silentAssertConnectedNetwork,
        assertConnectedNetwork,
        displayNetInfoModal,
        hideNetInfoModal,
        assertLedgerConnectivity,
      }}
    >
      {children}
      <NetInfoModal visible={isNetInfoModalDisplayed} onSubmit={() => hideNetInfoModal()} />
    </NetworkContext.Provider>
  )
}

export const useNetwork = () => useContext(NetworkContext)
