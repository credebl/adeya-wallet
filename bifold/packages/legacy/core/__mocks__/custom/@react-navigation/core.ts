const navigate = jest.fn()

const navigation = {
  navigate,
  setOptions: jest.fn(),
  getParent: jest.fn(() => ({
    navigate,
  })),
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

export { useNavigation, useIsFocused }
