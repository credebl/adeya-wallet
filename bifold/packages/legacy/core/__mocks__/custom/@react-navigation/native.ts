const useIsFocused = jest.fn()
const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

export { useIsFocused, useFocusEffect, createNavigatorFactory }
