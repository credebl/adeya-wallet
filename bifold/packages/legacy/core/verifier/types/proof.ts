import { ProofIdentifier } from '@aries-framework/core'

export interface MissingAttribute {
  name: string
}

export interface SharedAttribute {
  name: string
  value: string
  identifiers: ProofIdentifier
}

export interface SharedGroupedAttribute {
  name: string
  value: string
}

export interface SharedAttributesGroup {
  attributes: Array<SharedGroupedAttribute>
  identifiers: ProofIdentifier
}

export interface ResolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
  identifiers: ProofIdentifier
}

export interface UnresolvedPredicate {
  name: string
  predicateType: string
  predicateValue: number
}

export class ParsedIndyProof {
  public sharedAttributes: Array<SharedAttribute>
  public sharedAttributeGroups: Array<SharedAttributesGroup>
  public resolvedPredicates: Array<ResolvedPredicate>
  public unresolvedAttributes: Array<MissingAttribute>
  public unresolvedAttributeGroups: Array<Array<MissingAttribute>>
  public unresolvedPredicates: Array<UnresolvedPredicate>

  public constructor() {
    this.sharedAttributes = []
    this.sharedAttributeGroups = []
    this.resolvedPredicates = []
    this.unresolvedAttributes = []
    this.unresolvedAttributeGroups = []
    this.unresolvedPredicates = []
  }
}

export class CredentialSharedProofData {
  public sharedAttributes: Array<SharedAttribute>
  public sharedAttributeGroups: Array<SharedAttributesGroup>
  public resolvedPredicates: Array<ResolvedPredicate>

  public constructor() {
    this.sharedAttributes = []
    this.sharedAttributeGroups = []
    this.resolvedPredicates = []
  }
}

export type GroupedSharedProofDataItem = { data: CredentialSharedProofData; identifiers: ProofIdentifier }
export type GroupedSharedProofData = Map<string, GroupedSharedProofDataItem>
