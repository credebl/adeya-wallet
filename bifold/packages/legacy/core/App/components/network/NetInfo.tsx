import * as React from 'react'
import { useEffect } from 'react'
import { useTranslation } from 'react-i18next'
import Toast from 'react-native-toast-message'

import { useNetwork } from '../../contexts/network'

const NetInfo: React.FC = () => {
  const { silentAssertConnectedNetwork, assertLedgerConnectivity } = useNetwork()
  const { t } = useTranslation()

  const isConnected = silentAssertConnectedNetwork()

  useEffect(() => {
    if (isConnected) {
      assertLedgerConnectivity().then((status) => {
        if (status) {
          return
        }

        Toast.show({
          type: 'warn',
          autoHide: false,
          text1: t('NetInfo.LedgerConnectivityIssueMessage'),
        })
      })

      return
    }

    Toast.show({
      type: 'error',
      autoHide: true,
      text1: t('NetInfo.NoInternetConnectionTitle'),
    })
  }, [isConnected])

  return null
}

export default NetInfo
