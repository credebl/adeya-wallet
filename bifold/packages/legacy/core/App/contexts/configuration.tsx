import { IndyPoolConfig } from '@aries-framework/core'
import { createContext, ReducerAction, useContext } from 'react'

import { ProofRequestTemplate } from '../../verifier'
import { EmptyListProps } from '../components/misc/EmptyList'
import { RecordProps } from '../components/record/Record'
import OnboardingPages from '../screens/OnboardingPages'
import { ScanProps } from '../screens/Scan'
import { OCABundleResolver } from '../types/oca'
import { PINSecurityParams } from '../types/security'
import { SettingSection } from '../types/settings'

interface NotificationConfiguration {
  component: React.FC
  onCloseAction: (dispatch?: React.Dispatch<ReducerAction<any>>) => void
  title: string
  description: string
  buttonTitle: string
  pageTitle: string
}

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  homeContentView: React.FC
  credentialListHeaderRight: React.FC
  credentialListOptions: React.FC
  credentialEmptyList: React.FC<EmptyListProps>
  developer: React.FC
  OCABundleResolver: OCABundleResolver
  scan: React.FC<ScanProps>
  useBiometry: React.FC
  record: React.FC<RecordProps>
  PINSecurity: PINSecurityParams
  indyLedgers: IndyPoolConfig[]
  settings: SettingSection[]
  customNotification: NotificationConfiguration
  useCustomNotifications: () => { total: number; notifications: any }
  connectionTimerDelay?: number
  autoRedirectConnectionToHome?: boolean
  proofRequestTemplates?: Array<ProofRequestTemplate>
  enableTours?: boolean
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
