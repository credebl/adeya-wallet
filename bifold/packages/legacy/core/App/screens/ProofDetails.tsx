import type { StackScreenProps } from '@react-navigation/stack'

import { ProofExchangeRecord, ProofState } from '@aries-framework/core'
import { useAgent, useConnectionById, useProofById } from '@aries-framework/react-hooks'
import { useFocusEffect } from '@react-navigation/native'
import { StackNavigationProp } from '@react-navigation/stack'
import React, { useCallback, useEffect, useMemo, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { BackHandler, ScrollView, StyleSheet, Text, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'

import { ProofCustomMetadata, ProofMetadata, GroupedSharedProofDataItem, markProofAsViewed } from '../../verifier'
import CheckInCircle from '../assets/img/check-in-circle.svg'
import Button, { ButtonType } from '../components/buttons/Button'
import SharedProofData from '../components/misc/SharedProofData'
import { useTheme } from '../contexts/theme'
import { ProofRequestsStackParams, Screens } from '../types/navigators'
import { testIdWithKey } from '../utils/testable'

type ProofDetailsProps = StackScreenProps<ProofRequestsStackParams, Screens.ProofDetails>

interface VerifiedProofProps {
  record: ProofExchangeRecord
  isHistory?: boolean
  navigation: StackNavigationProp<ProofRequestsStackParams, Screens.ProofDetails>
}

interface UnverifiedProofProps {
  record: ProofExchangeRecord
}

const VerifiedProof: React.FC<VerifiedProofProps> = ({ record, navigation, isHistory }: VerifiedProofProps) => {
  const { t } = useTranslation()
  const { ColorPallet, TextTheme } = useTheme()

  const styles = StyleSheet.create({
    container: {
      flexGrow: 1,
    },
    header: {
      backgroundColor: ColorPallet.semantic.success,
      paddingHorizontal: 30,
      paddingVertical: 20,
    },
    headerTitleContainer: {
      flexDirection: 'row',
      justifyContent: 'flex-start',
      alignItems: 'center',
    },
    headerTitle: {
      marginHorizontal: 10,
      color: ColorPallet.grayscale.white,
      fontSize: 34,
      fontWeight: 'bold',
    },
    headerDetails: {
      color: ColorPallet.grayscale.white,
      marginVertical: 10,
      fontSize: 18,
    },
    descriptionContainer: {
      marginHorizontal: 30,
      marginVertical: 30,
    },
    descriptionText: {
      fontSize: 18,
      color: TextTheme.normal.color,
    },
    label: {
      fontWeight: 'bold',
    },
    content: {
      flexGrow: 1,
      marginHorizontal: 30,
      marginTop: 10,
    },
    footerButton: {
      margin: 20,
    },
  })

  const connection = useConnectionById(record.connectionId || '')
  const connectionLabel = useMemo(
    () => (connection ? connection?.alias || connection?.theirLabel : t('Verifier.ConnectionLessLabel')),
    [connection]
  )

  const [sharedProofDataItems, setSharedProofDataItems] = useState<GroupedSharedProofDataItem[]>([])

  const onSharedProofDataLoad = (data: GroupedSharedProofDataItem[]) => {
    setSharedProofDataItems(data)
  }

  const onGenerateNew = useCallback(() => {
    const metadata = record.metadata.get(ProofMetadata.customMetadata) as ProofCustomMetadata
    if (metadata.proof_request_template_id) {
      navigation.navigate(Screens.ProofRequesting, { templateId: metadata.proof_request_template_id })
    } else {
      navigation.navigate(Screens.ProofRequests, {})
    }
  }, [navigation])

  useEffect(() => {
    if (!connection || !isHistory) return
    navigation.setOptions({ title: connectionLabel })
  }, [connection])

  if (isHistory) {
    return (
      <ScrollView style={{ flexGrow: 1 }} testID={testIdWithKey('ProofDetailsHistoryView')}>
        <View style={styles.container}>
          {sharedProofDataItems.length > 0 && (
            <View style={styles.descriptionContainer}>
              <Text style={styles.descriptionText}>
                <Text style={styles.label}>{connectionLabel}</Text>{' '}
                {t('ProofRequest.ShareFollowingInformation', { count: sharedProofDataItems.length })}
              </Text>
            </View>
          )}
          <View style={styles.content}>
            <SharedProofData recordId={record.id} onSharedProofDataLoad={onSharedProofDataLoad} />
          </View>
        </View>
      </ScrollView>
    )
  }

  return (
    <ScrollView contentContainerStyle={{ flexGrow: 1 }} testID={testIdWithKey('ProofDetailsView')}>
      <View style={styles.container}>
        <View style={styles.header}>
          <View style={styles.headerTitleContainer}>
            <CheckInCircle {...{ height: 45, width: 45 }} />
            <Text style={styles.headerTitle}>{t('Verifier.InformationReceived')}</Text>
          </View>
          <Text style={styles.headerDetails}>
            {connectionLabel} {t('Verifier.InformationReceivedDetails')}
          </Text>
        </View>
        <View style={styles.content}>
          <SharedProofData recordId={record.id} />
        </View>
        <View style={styles.footerButton}>
          <Button
            title={t('Verifier.GenerateNewQR')}
            accessibilityLabel={t('Verifier.GenerateNewQR')}
            testID={testIdWithKey('GenerateNewQR')}
            buttonType={ButtonType.Primary}
            onPress={onGenerateNew}
          />
        </View>
      </View>
    </ScrollView>
  )
}

const UnverifiedProof: React.FC<UnverifiedProofProps> = ({ record }) => {
  const { t } = useTranslation()
  const { ColorPallet } = useTheme()

  const styles = StyleSheet.create({
    header: {
      flexGrow: 1,
      backgroundColor: ColorPallet.semantic.error,
      paddingHorizontal: 30,
      paddingVertical: 20,
    },
    headerTitleContainer: {
      flexDirection: 'row',
      justifyContent: 'flex-start',
      alignItems: 'center',
    },
    headerTitle: {
      marginHorizontal: 8,
      color: ColorPallet.grayscale.white,
      fontSize: 34,
      fontWeight: 'bold',
    },
    headerDetails: {
      color: ColorPallet.grayscale.white,
      marginVertical: 10,
      fontSize: 18,
    },
  })

  return (
    <View testID={testIdWithKey('UnverifiedProofView')}>
      <View style={styles.header}>
        <View style={styles.headerTitleContainer}>
          <Icon name="bookmark-remove" size={45} color={'white'} />
          {record.state === ProofState.Abandoned && (
            <Text style={styles.headerTitle}>{t('Verifier.PresentationDeclined')}</Text>
          )}
          {record.isVerified === false && (
            <Text style={styles.headerTitle}>{t('Verifier.ProofVerificationFailed')}</Text>
          )}
        </View>
      </View>
    </View>
  )
}

const ProofDetails: React.FC<ProofDetailsProps> = ({ route, navigation }) => {
  if (!route?.params) {
    throw new Error('ProofRequesting route prams were not set properly')
  }

  const { recordId, isHistory } = route?.params
  const record: ProofExchangeRecord = useProofById(recordId)
  const { agent } = useAgent()

  useEffect(() => {
    if (agent) markProofAsViewed(agent, record)
  }, [])

  useFocusEffect(
    useCallback(() => {
      const onBackPress = () => {
        if (route.params.isHistory) {
          navigation.goBack()
        } else {
          navigation.navigate(Screens.ProofRequests, {})
        }
        return true
      }

      BackHandler.addEventListener('hardwareBackPress', onBackPress)

      return () => BackHandler.removeEventListener('hardwareBackPress', onBackPress)
    }, [])
  )

  return (
    <SafeAreaView style={{ flexGrow: 1 }} edges={['left', 'right']}>
      {record.isVerified && <VerifiedProof record={record} isHistory={isHistory} navigation={navigation} />}
      {!record.isVerified && <UnverifiedProof record={record} />}
    </SafeAreaView>
  )
}

export default ProofDetails
