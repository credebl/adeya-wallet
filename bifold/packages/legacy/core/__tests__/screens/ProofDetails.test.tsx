import {
  INDY_PROOF_REQUEST_ATTACHMENT_ID,
  ProofExchangeRecord,
  ProofState,
  V1RequestPresentationMessage,
} from '@aries-framework/core'
import { Attachment, AttachmentData } from '@aries-framework/core/build/decorators/attachment/Attachment'
import { useProofById } from '@aries-framework/react-hooks'
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock'
import { useNavigation } from '@react-navigation/core'
import '@testing-library/jest-native/extend-expect'
import { act, cleanup, fireEvent, render, RenderAPI } from '@testing-library/react-native'
import React from 'react'

import { testIdWithKey } from '../../App/utils/testable'
import ProofDetails from '../../App/screens/ProofDetails'
import * as proofUtils from '../../verifier/utils/proof'
import { IndyProof } from 'indy-sdk-react-native'
import configurationContext from '../contexts/configuration'
import { NetworkProvider } from '../../App/contexts/network'
import { ConfigurationContext } from '../../App/contexts/configuration'

jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo)
jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})
// eslint-disable-next-line @typescript-eslint/no-empty-function
jest.mock('react-native-localize', () => {})
jest.useFakeTimers('legacy')
jest.spyOn(global, 'setTimeout')

const proof_request = {
  name: 'proof-request',
  version: '1.0',
  nonce: '1298236324864',
  requested_attributes: {
    attribute_1: {
      names: ['first_name', 'second_name'],
    },
  },
  requested_predicates: {},
}

const proof: IndyProof = {
  proof: {
    proofs: [],
    aggregated_proof: {
      c_list: [],
    },
  },
  requested_proof: {
    revealed_attr_groups: {
      attribute_1: {
        sub_proof_index: 0,
        values: {
          first_name: {
            raw: 'Aries',
            encoded: '',
          },
          second_name: {
            raw: 'Bifold',
            encoded: '',
          },
        },
      },
    },
    self_attested_attrs: {},
    unrevealed_attrs: {},
    // @ts-ignore
    predicates: {},
  },
  identifiers: [
    {
      schema_id: '7KuDTpQh3GJ7Gp6kErpWvM:2:Faber College:1.0.0',
      cred_def_id: '7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712167:latest',
    },
    {
      schema_id: '7KuDTpQh3GJ7Gp6kErpWvM:2:Passport:1.0.0',
      cred_def_id: '7KuDTpQh3GJ7Gp6kErpWvM:3:CL:712168:latest',
    },
  ],
}

const requestPresentationMessage = new V1RequestPresentationMessage({
  comment: 'some comment',
  requestPresentationAttachments: [
    new Attachment({
      id: INDY_PROOF_REQUEST_ATTACHMENT_ID,
      mimeType: 'application/json',
      data: new AttachmentData({
        json: proof_request,
      }),
    }),
  ],
})

const data = proofUtils.parseIndyProof(proof_request, proof)

describe('ProofDetails Component', () => {
  afterEach(() => {
    cleanup()
  })

  beforeEach(() => {
    jest.clearAllMocks()
  })

  const renderView = (params?: { recordId: string; isHistory: boolean }) => {
    return render(
      <ConfigurationContext.Provider value={configurationContext}>
        <NetworkProvider>
          <ProofDetails navigation={useNavigation()} route={{ params } as any} />
        </NetworkProvider>
      </ConfigurationContext.Provider>
    )
  }

  describe('with a verified proof record', () => {
    beforeEach(() => {
      jest.clearAllMocks()

      // @ts-ignore
      useProofById.mockReturnValue(testVerifiedProofRequest)
      // @ts-ignore
      jest.spyOn(proofUtils, 'getProofData').mockReturnValue(Promise.resolve(data))
    })

    const testVerifiedProofRequest = new ProofExchangeRecord({
      connectionId: undefined,
      threadId: requestPresentationMessage.id,
      state: ProofState.Done,
      protocolVersion: 'V1',
      isVerified: true,
    })

    const checkAttributes = async (tree: RenderAPI) => {
      const firstNameAttribute = await tree.findByText('first_name', { exact: true })
      const firstNameAttributeValue = await tree.findByText('Aries', { exact: true })
      const secondNameAttribute = await tree.findByText('second_name', { exact: true })
      const secondNameAttributeValue = await tree.findByText('Bifold', { exact: true })

      expect(firstNameAttribute).not.toBe(null)
      expect(firstNameAttributeValue).not.toBe(null)
      expect(secondNameAttribute).not.toBe(null)
      expect(secondNameAttributeValue).not.toBe(null)
    }

    test('renders correctly when history is true', async () => {
      const tree = renderView({ recordId: testVerifiedProofRequest.id, isHistory: true })

      await act(async () => null)
      expect(tree).toMatchSnapshot()

      const proofDetailsHistoryView = await tree.findByTestId(testIdWithKey('ProofDetailsHistoryView'))
      expect(proofDetailsHistoryView).not.toBe(null)

      await checkAttributes(tree)
    })

    test('renders correctly when history is false', async () => {
      const tree = renderView({ recordId: testVerifiedProofRequest.id, isHistory: false })

      await act(async () => null)
      expect(tree).toMatchSnapshot()

      const proofDetailsView = await tree.findByTestId(testIdWithKey('ProofDetailsView'))
      const generateNewButton = await tree.findByTestId(testIdWithKey('GenerateNewQR'))

      expect(proofDetailsView).not.toBe(null)
      expect(generateNewButton).not.toBe(null)

      await checkAttributes(tree)
    })

    test('Generate new QR code button should navigate correctly', async () => {
      const navigation = useNavigation()
      const tree = renderView({ recordId: testVerifiedProofRequest.id, isHistory: false })

      await act(async () => null)

      const generateNewButton = tree.getByTestId(testIdWithKey('GenerateNewQR'))
      expect(generateNewButton).not.toBeNull()

      fireEvent(generateNewButton, 'press')
      expect(navigation.navigate).toBeCalledWith('Proof Requests', {})
    })
  })

  describe('renders correctly with an unverified proof', () => {
    const testUnverifiedProofRequest = new ProofExchangeRecord({
      connectionId: '123',
      threadId: requestPresentationMessage.id,
      state: ProofState.Done,
      protocolVersion: 'V1',
      isVerified: false,
    })

    beforeEach(() => {
      jest.clearAllMocks()

      // @ts-ignore
      useProofById.mockReturnValue(testUnverifiedProofRequest)
    })

    test('Unverified proof view should be rendered correctly', async () => {
      const tree = renderView({ recordId: testUnverifiedProofRequest.id, isHistory: false })

      await act(async () => null)
      expect(tree).toMatchSnapshot()

      const unverifiedView = await tree.getByTestId(testIdWithKey('UnverifiedProofView'))
      expect(unverifiedView).not.toBeNull()
    })
  })
})
