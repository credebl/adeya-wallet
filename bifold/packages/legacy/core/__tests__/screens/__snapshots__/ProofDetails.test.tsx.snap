// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProofDetails Component renders correctly with an unverified proof Unverified proof view should be rendered correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <View
    testID="com.ariesbifold:id/UnverifiedProofView"
  >
    <View
      style={
        Object {
          "backgroundColor": "#D8292F",
          "flexGrow": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 45,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ó°ƒ†
        </Text>
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 34,
              "fontWeight": "bold",
              "marginHorizontal": 8,
            }
          }
        >
          Verifier.ProofVerificationFailed
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`ProofDetails Component with a verified proof record renders correctly when history is false 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    testID="com.ariesbifold:id/ProofDetailsView"
  >
    <View>
      <View
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#2E8540",
              "paddingHorizontal": 30,
              "paddingVertical": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              }
            }
          >
            <
              height={45}
              width={45}
            />
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 34,
                  "fontWeight": "bold",
                  "marginHorizontal": 10,
                }
              }
            >
              Verifier.InformationReceived
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "marginVertical": 10,
              }
            }
          >
            Verifier.ConnectionLessLabel
             
            Verifier.InformationReceivedDetails
          </Text>
        </View>
        <View
          style={
            Object {
              "flexGrow": 1,
              "marginHorizontal": 30,
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 10,
                  "marginBottom": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "minHeight": 247.5,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#5f0f7b",
                      },
                      Object {
                        "borderBottomLeftRadius": 10,
                        "borderTopLeftRadius": 10,
                        "width": 90,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/CardColor"
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 8,
                      "elevation": 5,
                      "height": 90,
                      "justifyContent": "center",
                      "left": -52.5,
                      "top": 37.5,
                      "width": 90,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 18,
                          "fontWeight": "normal",
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "#000000",
                          "fontSize": 45,
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    U
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginBottom": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontSize": 16,
                          "fontWeight": "normal",
                        }
                      }
                    >
                      first_name
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        }
                      }
                      testID="com.ariesbifold:id/AttributeValue"
                    >
                      Aries
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "marginBottom": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontSize": 16,
                          "fontWeight": "normal",
                        }
                      }
                    >
                      second_name
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        }
                      }
                      testID="com.ariesbifold:id/AttributeValue"
                    >
                      Bifold
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "margin": 20,
            }
          }
        >
          <View
            accessibilityLabel="Verifier.GenerateNewQR"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#42803E",
                "borderRadius": 4,
                "opacity": 1,
                "padding": 16,
              }
            }
            testID="com.ariesbifold:id/GenerateNewQR"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    false,
                  ]
                }
              >
                Verifier.GenerateNewQR
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`ProofDetails Component with a verified proof record renders correctly when history is true 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
    ]
  }
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flexGrow": 1,
      }
    }
    testID="com.ariesbifold:id/ProofDetailsHistoryView"
  >
    <View>
      <View
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 30,
              "marginVertical": 30,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Verifier.ConnectionLessLabel
            </Text>
             
            ProofRequest.ShareFollowingInformation
          </Text>
        </View>
        <View
          style={
            Object {
              "flexGrow": 1,
              "marginHorizontal": 30,
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 10,
                  "marginBottom": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "minHeight": 247.5,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#5f0f7b",
                      },
                      Object {
                        "borderBottomLeftRadius": 10,
                        "borderTopLeftRadius": 10,
                        "width": 90,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/CardColor"
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#FFFFFF",
                      "borderRadius": 8,
                      "elevation": 5,
                      "height": 90,
                      "justifyContent": "center",
                      "left": -52.5,
                      "top": 37.5,
                      "width": 90,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 18,
                          "fontWeight": "normal",
                        },
                        Object {
                          "alignSelf": "center",
                          "color": "#000000",
                          "fontSize": 45,
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    U
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "paddingBottom": 10,
                      "paddingTop": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginBottom": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontSize": 16,
                          "fontWeight": "normal",
                        }
                      }
                    >
                      first_name
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        }
                      }
                      testID="com.ariesbifold:id/AttributeValue"
                    >
                      Aries
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "marginBottom": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontSize": 16,
                          "fontWeight": "normal",
                        }
                      }
                    >
                      second_name
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#000000",
                          "fontSize": 18,
                          "fontWeight": "bold",
                        }
                      }
                      testID="com.ariesbifold:id/AttributeValue"
                    >
                      Bifold
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
